packages/block-editor/src/components/list-view/index.js

```diff
@@ -65,6 +65,7 @@ export const BLOCK_LIST_ITEM_HEIGHT = 36;
  * @param {?ComponentType} props.blockSettingsMenu Optional more menu substitution. Defaults to the standard `BlockSettingsDropdown` component.
  * @param {string}         props.rootClientId      The client id of the root block from which we determine the blocks to show in the list.
  * @param {string}         props.description       Optional accessible description for the tree grid component.
+ * @param {string}         props.onSelect          Optional callback to be invoked when a block is selected.
```

Comment:
```suggestion
 * @param {?Function}      props.onSelect          Optional callback to be invoked when a block is selected.
```

Tiny nit: optional function.

Reply:
Also, should we flag that the callback receives the block object that was selected?